core/log_storage.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+# Class provides storage of logs
+class LogStorage < Out
+  # @param configuration [Session] configuration object that can silence the output
+  def initialize(configuration)
+    super(configuration)
+    @logs = []
+  end
+
+  def out(string)
+    return if string.nil?
+
+    @logs.push(string)
+  end
+
+  def print_to_stdout
+    @logs.each { |log_line| @stream.puts(log_line) }
+  end
+
+  private
```

Comment:
Make print_line the protected method in Out class, so it would be obvious why you create the print_line method in this class.